restapi/src/main/java/io/stargate/web/docsapi/service/query/DocumentSearchService.java

```diff
@@ -215,6 +221,36 @@
             });
   }
 
+  private Flowable<RawDocument> documentTtl(
+      QueryExecutor queryExecutor,
+      DocsApiConfiguration configuration,
+      String keyspace,
+      String collection,
+      String documentId) {
+    // prepare first
+    return RxUtils.singleFromFuture(
+            () -> {
+              DataStore dataStore = queryExecutor.getDataStore();
+
+              DocumentTtlQueryBuilder queryBuilder = new DocumentTtlQueryBuilder(documentId);
+              BuiltQuery<? extends BoundQuery> query =
+                  queryBuilder.buildQuery(dataStore::queryBuilder, keyspace, collection);
+
+              return dataStore.prepare(query);
+            })
+        .cache()
+        .flatMapPublisher(
+            prepared -> {
+              BoundQuery boundQuery = prepared.bind();
+              return queryExecutor.queryDocs(
+                  boundQuery,
+                  configuration.getMaxStoragePageSize(),
+                  false,
+                  null,
+                  ExecutionContext.NOOP_CONTEXT);
```

Comment:
why noop context here, don't we want to record this to the context as well? then people can see what's really going on?